views/js/pciCreator/dev/audioRecordingInteraction/runtime/js/recorder.js

```diff
@@ -0,0 +1,286 @@
+            if (MediaRecorder.isTypeSupported('audio/webm;codecs=opus')) {
+                recorderOptions.mimeType = 'audio/webm;codecs=opus';
+            } else if (MediaRecorder.isTypeSupported('audio/ogg;codecs=opus')) {
+                recorderOptions.mimeType = 'audio/ogg;codecs=opus';
+            } else if (MediaRecorder.isTypeSupported('audio/webm')) {
+                recorderOptions.mimeType = 'audio/webm';
+            } else if (MediaRecorder.isTypeSupported('audio/ogg')) {
+                recorderOptions.mimeType = 'audio/ogg';
+            }
+        }
+
+        recorder = {
+            /**
+             * Set recorder state
+             * @param {String} newState
+             * @private
+             */
+            _setState: function _setState(newState) {
+                state = newState;
+                this.trigger('statechange');
+                this.trigger(newState);
+            },
+
+            /**
+             * Check the current state
+             * @param {String} queriedState
+             * @returns {Boolean}
+             */
+            is: function is(queriedState) {
+                return (state === queriedState);
+            },
+
+            /**
+             * Initialise the MediaRecorder stream
+             * @returns {Promise}
+             */
+            init: function init() {
+                var self = this;
+
+                return navigator.mediaDevices.getUserMedia({ audio: true })
+                    .then(function(stream) {
+                        mediaRecorder = new MediaRecorder(stream, recorderOptions);
+                        mimeType = mediaRecorder.mimeType;
+
+                        initAnalyser(stream);
+
+                        self._setState(recorderStates.IDLE);
+
+                        // save chunks of the recording
+                        mediaRecorder.ondataavailable = function ondataavailable(e) {
+                            chunks.push(e.data);
+                        };
+
+                        // stop record callback
+                        mediaRecorder.onstop = function onstop() {
+                            var blob,
+                                durationMs;
+
+                            self.trigger('stop');
+
+                            if (! self.cancelled) {
+                                blob = new Blob(chunks, { type: mimeType });
+                                durationMs = new window.Date().getTime() - startTimeMs;
+                                self.trigger('recordingavailable', [blob, durationMs]);
+
+                            }
+                            cancelAnimationFrame(timerId);
+
+                            self.trigger('levelUpdate', [0]);
+                            self._setState(recorderStates.IDLE);
+
+                            chunks = [];
+                        };
+                    })
+                    .catch(function(err) {
+                        throw err;
```

Comment:
well... this is weird... If the promise is rejected, that will throw an error, and... the promise will be rejected again... So the code won't do more.
What was the idea?